src/mbgl/text/placement.cpp

```diff
@@ -1338,6 +1360,38 @@ bool TilePlacement::stickToFirstVariableAnchor(const CollisionBox& box,
     return collisionIndex.intersectsTileEdges(box, shift, posMatrix, textPixelRatio, *tileBorders);
 }
 
+void TilePlacement::newSymbolPlaced(const SymbolInstance& symbol,
+                                    const JointPlacement& placement,
+                                    style::SymbolPlacementType placementType,
+                                    const std::vector<ProjectedCollisionBox>& textBoxes,
+                                    const std::vector<ProjectedCollisionBox>& iconBoxes) {
+    if (!collectData || placementType != style::SymbolPlacementType::Point) return;
+
+    optional<mapbox::geometry::box<float>> textCollisionBox;
+    if (!textBoxes.empty()) {
+        assert(textBoxes.size() == 1u);
+        auto& box = textBoxes.front();
+        assert(box.isBox());
+        textCollisionBox = box.box();
+    }
+    optional<mapbox::geometry::box<float>> iconCollisionBox;
+    if (!iconBoxes.empty()) {
+        assert(iconBoxes.size() == 1u);
+        auto& box = iconBoxes.front();
+        assert(box.isBox());
+        iconCollisionBox = box.box();
+    }
+    PlacedSymbolData symbolData{symbol.key,
+                                symbol.dataFeatureIndex,
+                                textCollisionBox,
+                                iconCollisionBox,
+                                placement.text,
+                                placement.icon,
+                                onlyLabelsIntersectingTileBorders,
+                                collisionIndex.getViewportPadding()};
+    placedSymbolsData.emplace_back(std::move(symbolData));
```

Comment:
Is it ever cleared or needs to be cleared? What if we move it out to a client when getPlacedSymbolsData is called?

Reply:
it's cleared at `placeLayers()` call.